tests/phpunit/includes/abstract-testcase.php

```diff
@@ -1101,6 +1101,31 @@ public function assertQueryTrue( ...$prop ) {
 		}
 	}
 
+	/**
```

Comment:
Functionally, nothing. However, every release, we sift through potentially hundreds of `assertEquals()` to use stricter assertions instead. By having a clearly named wrapper, we can reduce workload each release. In addition, `assertSimilarObject()` makes it clear what is being tested for those reading a test method, whereas `assertEquals()` is just too ambiguous.

While this isn't required, there was some support for implementing this wrapper method to make life easier going forward. I'll drop a link to this in a bit.

In terms of why `assertEquals()` is appropriate here, `assertSame()` can't be used for objects, as their IDs are different and the assertion will always fail. `assertObjectEquals()` is only for value objects.

Reply:
See [this comment](https://core.trac.wordpress.org/ticket/53364#comment:6) and Sergey's reply.

The idea is that having the wrapper method would allow for the possibility of banning `assertEquals()` in `tests/phpunit/tests/`, to encourage stricter testing and reduce our workload (such as this PR)

Reply:
I like the idea of having a linting check that fails if an occurrences of `assertEquals()` are found in the tests. That's along what I had in mind.

It looks like there is `assertEqualsCanonicalizing()` in PHPUnit ([docs](https://phpunit.readthedocs.io/en/9.5/assertions.html?highlight=assertEqualsCanonicalizing#assertequalscanonicalizing)). But I'm not sure if this will work for all scenarios.

Could you take a look at that?

Reply:
There also may be shortcomings of that method that are not immediately apparent. So not saying we should switch these now without further discussion.